kompendium-oas/src/test/kotlin/io/bkbn/kompendium/oas/XmlSchemaTest.kt

```diff
@@ -0,0 +1,203 @@
+package io.bkbn.kompendium.oas
+
+import io.bkbn.kompendium.oas.component.Components
+import io.bkbn.kompendium.oas.schema.ArraySchema
+import io.bkbn.kompendium.oas.schema.ComponentSchema
+import io.bkbn.kompendium.oas.schema.ObjectSchema
+import io.bkbn.kompendium.oas.schema.SimpleSchema
+import io.kotest.core.spec.style.DescribeSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.serialization.json.Json
+
+private val encoder = Json {
+  prettyPrint = true
+  encodeDefaults = true
+  explicitNulls = false
+}
+
+class XmlSchemaTest : DescribeSpec({
+    describe("XmlSchema") {
+      it("name replacement") {
+        val animals = SimpleSchema("string", xml = SimpleSchema.Xml(name = "animal"))
```

Comment:
how do you envision this being leveraged by an end user? Would they need to declare custom type overrides for every `Xml` object that they want to declare? Or should there be some way for users to override in the object itself?

Reply:
by end user I assume you mean the developer creating the OpenApiSpec?  If so yes they would need to specify it explicitly - although as I'm looking through my usecases it's always due to kotlinx-serialization @SerialName("something else")  so that could potentially be leveraged automatically somehow?  I'm on ktor 2.0 so I'm only using the `oas` library and didn't look too far into the others.

Reply:
Are you rolling your own docs using that module? 👀

Reply:
yes using ktor 2.0 routes

Reply:
neat 👍

Reply:
I noticed this not being supported when I turned on xml serialization in my ktor 2.0 app and swagger-ui said in some cases A) it couldn't generate an example and in other cases B) the example was not the same as the kotlinx-serialization xml

Reply:
yea, I was hesitant to support ktor 2 since there have been rumors that Jetbrains is working on an official openapi doc plugin, but now I'm starting to think that it may be better to just go ahead and port kompendium to support ktor 2.  

I've started to poke around in what that would entail recently.... 

unfortunately it would be a lot of work, since while I'm at it I would want to fix a bunch of the bugs that are currently present in Kompendium today.

Reply:
There is an issue tracking it https://github.com/bkbnio/kompendium/issues/96 along with a tentative project https://github.com/orgs/bkbnio/projects/5 of what would be required